overlord/devicestate/devicestate_remodel_test.go

```diff
@@ -2024,6 +2024,11 @@ func (s *deviceMgrRemodelSuite) TestRemodelUC20RequiredSnapsAndRecoverySystem(c
 				"id":       snaptest.AssertedSnapID("new-required-snap-2"),
 				"presence": "required",
 			},
+			map[string]interface{}{
```

Comment:
Adding this results in the aforementioned panic without the code changes in `remodelTasks`.